app/scripts/background.js

```diff
@@ -645,7 +667,7 @@ export async function loadStateFromPersistence() {
   persistenceManager.setMetadata(versionedData.meta);
 
   // write to disk
-  persistenceManager.set(versionedData.data);
+  await persistenceManager.set(versionedData.data);
```

Comment:
I almost wonder if this will fix a bunch of other bugs. Wait weren't waiting for the database to be saved before using the data. 🤔